extension/lib/garbage-collector.js

```diff
@@ -80,12 +37,40 @@ const reporter = EventEmitter.compose({
     if ('timestamp' in data)
       data['timestamp'] = Math.round(data['timestamp'] / 1000);
 
-    // Once the console listener can be removed, we can emit directly
-    timers.setTimeout(function (aScope) {
-      aScope._emit(type, data);
-    }, 0, this);
+    emit(reporter, type, data);
+  }
+}
+
+function moduleUnload() {
+    off(reporter);
+    Services.obs.removeObserver(gcCcObserver, config.application.topic_cc_statistics, false);
+    Services.obs.removeObserver(gcCcObserver, config.application.topic_gc_statistics, false);
+}
+
+unload.ensure(this, 'moduleUnload');
+
+function _enable() {
+  var modifiedPrefs = JSON.parse(prefs.get(config.preferences.modified_prefs,
+                                            "{}"));
+  if (!modifiedPrefs.hasOwnProperty(PREF_GC_NOTIFICATIONS)) {
+    modifiedPrefs[PREF_GC_NOTIFICATIONS] = prefs.get(PREF_GC_NOTIFICATIONS);
   }
-})();
+
+  prefs.set(PREF_GC_NOTIFICATIONS, true);
+  prefs.set(config.preferences.modified_prefs, JSON.stringify(modifiedPrefs));
+  _isEnabled = true;
+}
+
+// Ensure GC/CC observer and console messages preference is enabled
+_isEnabled = prefs.get(PREF_GC_NOTIFICATIONS);
+if (!_isEnabled)
+  _enable();
+
+reporter.on = on.bind(null, reporter);
+reporter.off = off.bind(null, reporter);
+
+Services.obs.addObserver(gcCcObserver, config.application.topic_cc_statistics, false);
+Services.obs.addObserver(gcCcObserver, config.application.topic_gc_statistics, false);
```

Comment:
Please do all the initialization code in an `init()` method of the gcObserver. I would kill the extra `Cc` from its name because it also handles garbage collection and not only cycle collection.